File: C# OOP/Exams/C# OOP Exam - 16 August 2020/01. Structure_Skeleton/OnlineShop-Skeleton/OnlineShop/Models/Products/Components/PowerSupply.cs
﻿
namespace OnlineShop.Models.Products.Components
{
    public class PowerSupply : Component
    {
        private const double POWER_SUPPLY_MULTIPLIER = 1.05;

        public PowerSupply(int id, string manufacturer, string model, decimal price, double overallPerformance, int generation) : base(id, manufacturer, model, price, overallPerformance, generation)
        {
            //this.OverallPerformance *= POWER_SUPPLY_MULTIPLIER;
        }
        public override double OverallPerformance => base.OverallPerformance * POWER_SUPPLY_MULTIPLIER;
    }
}

File: C# OOP/Exams/Java OOP Exam 16 August 2020/OnlineShop/src/onlineShop/models/products/components/PowerSupply.java
package onlineShop.models.products.components;

public class PowerSupply extends BaseComponent{
    private final static double PS_MULTIPLIER = 1.05;
    public PowerSupply(int id, String manufacturer, String model, double price, double overallPerformance, int generation) {
        super(id, manufacturer, model, price, overallPerformance * PS_MULTIPLIER, generation);
    }
}
